pycytominer/tests/test_cyto_utils/sqlite/conftest.py

```diff
@@ -0,0 +1,213 @@
+"""
+Pytest conftest file for sqlite series of tests.
+"""
+
+import os
+import tempfile
+from typing import Dict, List
+
+import pytest
+from pycytominer.cyto_utils.sqlite.convert import sql_select_distinct_join_chunks
+from pycytominer.cyto_utils.sqlite.meta import collect_columns
+from sqlalchemy import create_engine
+from sqlalchemy.engine.base import Engine
+
+
+@pytest.fixture
+def database_engine_for_testing() -> Engine:
+    """
+    A database engine for testing as a fixture.
+    """
+
+    # get temporary directory
+    tmpdir = tempfile.gettempdir()
+
+    # create a temporary sqlite connection
+    sql_path = f"sqlite:///{tmpdir}/test_sqlite.sqlite"
+
+    engine = create_engine(sql_path)
+
+    # statements for creating database with simple structure
+    create_stmts = [
+        "drop table if exists tbl_a;",
+        """
+        create table tbl_a (
+        col_integer INTEGER NOT NULL
+        ,col_text TEXT
+        ,col_blob BLOB
+        ,col_real REAL
+        );
+        """,
+        "drop table if exists tbl_b;",
+        """
+        create table tbl_b (
+        col_integer INTEGER
+        ,col_text TEXT
+        ,col_blob BLOB
+        ,col_real REAL
+        );
+        """,
+    ]
+
+    insert_vals = [1, "sample", b"sample_blob", 0.5]
+
+    with engine.begin() as connection:
+        for stmt in create_stmts:
+            connection.execute(stmt)
+
+        # insert statement with some simple values
+        # note: we use SQLAlchemy's parameters to insert data properly, esp. BLOB
+        connection.execute(
```

Comment:
is this the dreaded sql injection vulnerability? Do we want to test this way?

Reply:
Thank you for this comment. I leaned towards a simple approach with these test inserts over [SQLAlchemy insert compilation](https://docs.sqlalchemy.org/en/14/tutorial/data_insert.html#inserting-rows-with-core) to help increase understandability and presuming a low risk vector for the non-public-facing SQLite databases.

I can understand wanting to keep things secure, protecting against future scenarios when the databases may be public-facing, or taking a more SQLAlchemy-ORM approach here. The cost might be greater complexity for the codebase for SQLite work. Do you think we should change these inserts (and related) towards this effect?

Reply:
I'm in favor of simplicity. Perhaps the current branch and testing protections are sufficient. Maybe best to proceed as-is?